### PR TITLE
Fix table initial load

### PR DESCRIPTION
**Pull request by MichelML**
tableinitialload

As shown in the above gif, the initial load is currently not working properly. The blankslate pops up while the initial results are retrieved. The initial loading behavior should be displayed instead. This PR fixes the issue.    
  
also adressed the following issue:  https://github.com/coveo/react-vapor/issues/621

### CODE REVIEW

File: src/components/tables/Table.tsx [hunk 1]
```diff
@@ -141,7 +141,7 @@ export class Table extends React.Component<ITableProps> {
     }
 
     componentDidUpdate() {
-        if (this.isInitialLoad && !_.isUndefined(this.props.tableCompositeState.data)) {
+        if (this.isInitialLoad && JSON.stringify(this.props.tableCompositeState.data) !== JSON.stringify(DEFAULT_TABLE_DATA)) {
```
Comment (gdostie) [inline src/components/tables/Table.tsx]: `DEFAULT_TABLE_DATA` --> `this.props.initialTableData`

File: src/components/tables/Table.tsx [hunk 2]
```diff
@@ -264,4 +263,11 @@ export class Table extends React.Component<ITableProps> {
                 </tbody>
             );
     }
+
+    private shouldShowTableBody(): boolean {
+        const tableData = this.props.tableCompositeState.data || this.props.initialTableData;
+
+        return !this.props.tableCompositeState.isInError
+            && !!(tableData.displayedIds.length || this.props.tableCompositeState.isLoading || this.isInitialLoad);
```
Comment (gdostie) [inline src/components/tables/Table.tsx]: `!_.isEmpty(tableData.displayedIds)` is safer than `tableData.displayedIds.length`,but maybe `tableData.displayedIds` never is undefined in that case thought. Very minor, your call.

File: src/components/tables/Table.tsx (hunk 1 above)
Comment (MichelML) [reply src/components/tables/Table.tsx]: we really want DEFAULT_TABLE_DATA here

File: src/components/tables/Table.tsx (hunk 2 above)
Comment (MichelML) [reply src/components/tables/Table.tsx]: will never be undefined 👍

Comment (MichelML) [reply src/components/tables/Table.tsx]: but you're right 😄

**Pull request merged**